### PR TITLE
Fix `x.py install`

### PR DESCRIPTION
**Pull request by Hoverbear**
@pietroalbini noticed that `x.py install` was broken due to some of the changes we'd made.

This is largely due to us packaging our tarballs slightly differently and not generating an `install.sh`.

This PR creates a new ferrocene specific install step for each package that directly renames the tarball outputs to their configured destinations.

It supports all the `config.toml` options as well as `DESTDIR`.

### Testing

* [ ] Test `x.py install` with a `install.prefix` set in `config.toml`
* [ ] Test `x.py install` with any number of `install.*dir` set in `config.toml`
* [ ] Test `x.py install` with a `install.prefix` and any number of `install.*dir` set in `config.toml`
* [ ] Test `x.py install` with the `DESTDIR` env set
* [ ] Test `x.py install` with the `DESTDIR` env set and a `install.prefix` set in `config.toml`
* [ ] Test `x.py install` with the `DESTDIR` env set and any number of `install.*dir` set in `config.toml`
* [ ] Test `x.py install` with the `DESTDIR` env set and a `install.prefix` set in `config.toml` and any number of `install.*dir` set in `config.toml`

### CODE REVIEW

File: src/bootstrap/src/ferrocene/install.rs [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+            }
+
+            fn run($sel, $builder: &Builder<'_>) {
+                $run_item
+            }
+        })+
+    }
+}
+
+install!((self, builder, _config),
```
Comment (pietroalbini) [inline src/bootstrap/src/ferrocene/install.rs:141]: If we want to provide support, we should replace this with the list of components we want people to actually be able to install.

**Review by pietroalbini — CHANGES_REQUESTED** [review]

File: src/bootstrap/src/ferrocene/install.rs (hunk 1 above)
Comment (Hoverbear) [reply src/bootstrap/src/ferrocene/install.rs:141]: Oh, ya! This should have like `flip-link` etc. Are there any I should remove? 🤔

**Comment by Hoverbear** [comment]
Looks like the `ferrocene-docs` job has some broken links...

File: src/bootstrap/src/ferrocene/install.rs (hunk 1 above)
Comment (pietroalbini) [reply src/bootstrap/src/ferrocene/install.rs:141]: We should probably remove the upstream docs.

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**